test/Microsoft.Health.Fhir.Shared.Tests.Integration/Persistence/SqlServerSchemaUpgradeTests.cs

```diff
@@ -274,7 +275,36 @@ private bool CompareDatabaseSchemas(string databaseName1, string databaseName2)
                 }
             }
 
+            // if TransactionCheckWithInitialiScript(which has current version as x-1) is not updated with the new x version then x.sql will have a wrong version inserted into SchemaVersion table
+            // this will cause an entry like (x-1, started) and (x, completed) at the end of of the transaction in fullsnapshot database (testConnectionString1)
+            // If any schema version is in started state then there might be a problem
+            if (SchemaVersionInStartedState(testConnectionString1).Result || SchemaVersionInStartedState(testConnectionString2).Result)
+            {
+                // if the test hits below statement then there is a possibility that TransactionCheckWithInitialiScript is not updated with the new version
+                unexpectedDifference = true;
+            }
+
             return !unexpectedDifference;
         }
+
+        public async Task<bool> SchemaVersionInStartedState(string connectionString)
+        {
+            using (SqlConnection sqlConnection = new SqlConnection(connectionString))
+            using (SqlCommand checkSchemaVersionCommand = sqlConnection.CreateCommand())
+            {
+                if (sqlConnection.State != ConnectionState.Open)
+                {
+                    await sqlConnection.OpenAsync(CancellationToken.None);
+                }
+
+                checkSchemaVersionCommand.CommandText = "SELECT count(*) FROM SchemaVersion where Status = 'started'";
+                if ((int?)await checkSchemaVersionCommand.ExecuteScalarAsync(CancellationToken.None) == 1)
```

Comment:
Should this be >= instead of ==?

Reply:
Yes, you are right! Updated with latest commit